src/helpers/await.ts

```diff
@@ -0,0 +1,62 @@
+export type Awaitable<T> = T | PromiseLike<T>;
```

Comment:
You can mostly skip this file. This is a series of helper interfaces and functions for dealing with promises.